### PR TITLE
Feature/1130 send log events

### PR DESCRIPTION
**Pull request by jonase**

### CODE REVIEW

**Comment by iperdomo** [comment]
## Checklist

* [ ] Test plan
* [ ] Copyright header
* [ ] Code formatting

File: Config/instanceConfigurator/README.md [hunk 1]
```diff
@@ -49,6 +49,9 @@ usage: org.akvo.flow.InstanceConfigurator
  -et,--emailTo <arg>        Recipient email of error notifications
  -fs,--flowServices <arg>   FLOW Services url, e.g.
                             http://services.akvoflow.org
+ -en,--eventNotification <arg>
+                            FLOW services event notification endpoint, e.g.
+                            http://services.akvoflow.org:3030/event_notification
```
Comment (jonase) [inline Config/instanceConfigurator/README.md]: We will need to add the emitChangeEvents property here as well.

File: scripts/data/src/org/akvo/gae/remoteapi/SurveyedLocaleFix.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+ * @author jonasenlund
+ *
+ */
+public class SurveyedLocaleFix implements Process {
+
+    final static Map<Long, Long> formIdToSurveyIdCache = new HashMap<>();
+
+    @Override
+    public void execute(DatastoreService ds, String[] args) throws Exception {
+        // TODO Auto-generated method stub
```
Comment (jonase) [inline scripts/data/src/org/akvo/gae/remoteapi/SurveyedLocaleFix.java]: @jonase remove this line!

File: GAE/war/WEB-INF/queue.xml [hunk 3]
```diff
@@ -24,7 +24,7 @@
           <task-retry-limit>10</task-retry-limit>
           <min-backoff-seconds>10</min-backoff-seconds>
         </retry-parameters>
-	</queue>
+    </queue>
```
Comment (jonase) [inline GAE/war/WEB-INF/queue.xml]: bad indent?

File: scripts/data/src/org/akvo/gae/remoteapi/ExportDataToEventLog.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                        }
+                    });
+
+            Iterable<List<Map<String, Object>>> eventBatches = Iterables
+                    .partition(events, BATCH_SIZE);
+
+            for (List<Map<String, Object>> eventBatch : eventBatches) {
+                System.out.println("  Batch #" + batch);
+                batch++;
+                EventUtils.sendEvents("http://flowdev1.akvo.org:3030/events/"
```
Comment (jonase) [inline scripts/data/src/org/akvo/gae/remoteapi/ExportDataToEventLog.java]: this url should not be hard-coded

File: GAE/src/org/akvo/flow/events/EventLogger.java [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+            m.writeValue(w, event);
+            BaseDAO<EventQueue> eventDao = new BaseDAO<EventQueue>(EventQueue.class);
+            EventQueue eventQ = new EventQueue();
+            eventQ.setCreatedDateTime(timestamp);
+            eventQ.setPayload(w.toString());
+            eventDao.save(eventQ);
+            notifyLog();
+        } catch (Exception e) {
+            logger.log(Level.SEVERE, "could not store " + event.get("eventType")
+                    + " event. Error: " + e.toString());
```
Comment (iperdomo) [inline GAE/src/org/akvo/flow/events/EventLogger.java]: We always want the call that uses, the `Throwable` as second parameter, as we want the full stack trace

File: GAE/src/org/akvo/flow/events/EventLogger.java [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+
+            // create event
+            Map<String, Object> event = newEvent(SystemProperty.applicationId.get(),
+                    types.action + actionType, eventEntity, eventContext);
+
+            // store it
+            storeEvent(event, timestamp);
+
+        } catch (Exception e) {
+            logger.log(Level.SEVERE, "Could not handle datastore put event: " + e.getMessage());
```
Comment (iperdomo) [inline GAE/src/org/akvo/flow/events/EventLogger.java]: idem ^^^

File: GAE/src/org/akvo/flow/events/EventLogger.java [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+
+            // create event
+            Map<String, Object> event = newEvent(SystemProperty.applicationId.get(),
+                    types.action + Action.DELETED, eventEntity, eventContext);
+
+            // store it
+            storeEvent(event, timestamp);
+
+        } catch (Exception e) {
+            logger.log(Level.SEVERE, "Could not handle datastore delete event: " + e.getMessage());
```
Comment (iperdomo) [inline GAE/src/org/akvo/flow/events/EventLogger.java]: idem ^^^

File: Config/instanceConfigurator/src/org/akvo/flow/InstanceConfigurator.java [hunk 8]
```diff
@@ -76,6 +76,8 @@ public static void main(String[] args) throws Exception {
         String gaeId = cli.getOptionValue("gae");
         String outFolder = cli.getOptionValue("o");
         String flowServices = cli.getOptionValue("fs");
+        String eventNotification = cli.getOptionValue("en");
+        String enableChangeEvents = cli.getOptionValue("ce");
```
Comment (iperdomo) [inline Config/instanceConfigurator/src/org/akvo/flow/InstanceConfigurator.java]: Do we want to fallback this option to `false` if the user doesn't pass any value? its optional parameter

Comment (jonase) [inline Config/instanceConfigurator/src/org/akvo/flow/InstanceConfigurator.java]: @iperdomo done

**Comment by iperdomo** [comment]
:+1: looks good

**Pull request merged**